packages/react-native-reanimated/Common/cpp/reanimated/CSS/interpolation/filters/FilterStyleInterpolator.cpp

```diff
@@ -0,0 +1,306 @@
+
+FilterStyleInterpolator::FilterStyleInterpolator(
+    const PropertyPath &propertyPath,
+    const std::shared_ptr<FilterOperationInterpolators> &interpolators,
+    const std::shared_ptr<ViewStylesRepository> &viewStylesRepository)
+    : PropertyInterpolator(propertyPath, viewStylesRepository), interpolators_(interpolators) {}
+
+folly::dynamic FilterStyleInterpolator::getStyleValue(const std::shared_ptr<const ShadowNode> &shadowNode) const {
+  return viewStylesRepository_->getStyleProp(shadowNode->getTag(), propertyPath_);
+}
+
+folly::dynamic FilterStyleInterpolator::getResetStyle(const std::shared_ptr<const ShadowNode> &shadowNode) const {
+  auto styleValue = getStyleValue(shadowNode);
+
+  if (!styleValue.isArray()) {
+    return convertOperationsToDynamic(defaultStyleValue_);
+  }
+
+  return styleValue;
+}
+
+folly::dynamic FilterStyleInterpolator::getFirstKeyframeValue() const {
+  return convertOperationsToDynamic(keyframes_.front()->fromOperations.value_or(defaultStyleValue_));
+}
+
+folly::dynamic FilterStyleInterpolator::getLastKeyframeValue() const {
+  return convertOperationsToDynamic(keyframes_.back()->toOperations.value_or(defaultStyleValue_));
+}
+
+bool FilterStyleInterpolator::equalsReversingAdjustedStartValue(const folly::dynamic &propertyValue) const {
+  const auto propertyOperations = parseFilterOperations(propertyValue);
+
+  if (!reversingAdjustedStartValue_.has_value()) {
+    return !propertyOperations.has_value();
+  } else if (!propertyOperations.has_value()) {
+    return false;
+  }
+
+  const auto &reversingAdjustedOperationsValue = reversingAdjustedStartValue_.value();
+  const auto &propertyOperationsValue = propertyOperations.value();
+
+  if (reversingAdjustedOperationsValue.size() != propertyOperationsValue.size()) {
+    return false;
+  }
+
+  for (size_t i = 0; i < reversingAdjustedOperationsValue.size(); ++i) {
+    if (*reversingAdjustedOperationsValue[i] != *propertyOperationsValue[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+folly::dynamic FilterStyleInterpolator::interpolate(
+    const std::shared_ptr<const ShadowNode> &shadowNode,
+    const std::shared_ptr<KeyframeProgressProvider> &progressProvider,
+    const double) const {
+  const auto currentIndex = getIndexOfCurrentKeyframe(progressProvider);
+
+  // Get or create the current keyframe
+  auto keyframe = keyframes_[currentIndex];
+  if (!keyframe->fromOperations.has_value() || !keyframe->toOperations.has_value()) {
+    // If the value is nullopt, we would have to read it from the view style
+    // and build the keyframe again
+    const auto fallbackValue = getFallbackValue(shadowNode);
+    keyframe = createFilterKeyframe(
+        keyframe->fromOffset,
+        keyframe->toOffset,
+        keyframe->fromOperations.value_or(fallbackValue),
+        keyframe->toOperations.value_or(fallbackValue));
+  }
+
+  if (!keyframe->isDiscrete) {
```

Comment:
```suggestion
  if (keyframe->isDiscrete) {
```

I think it should be the opposite. When the value is discrete, that means it cannot be numerically interpolated and we want to fall back to the immediate value change.